src/server/db_slice.h

```diff
@@ -374,7 +374,7 @@ class DbSlice {
   DbTableArray db_arr_;
 
   // Used in temporary computations in Acquire/Release.
-  mutable absl::flat_hash_set<std::string_view> uniq_keys_;
+  mutable absl::flat_hash_set<std::string> uniq_keys_;
```

Comment:
I didn't add it, just made it `mutable` (and also set it in a few places where it wasn't set before).
We actually use this for computing which keys to acquire/release (and have used it before I touched it :). But the thing is, we only use it for temporary strings and it could have been local, but as an optimization we reuse it to reduce allocations. I piggybacked on this to also return it for testing purposes.

---

Has anything changed to make this not work anymore?

Reply:
My bad, we use it inside transaction only for removing duplicates from the lock keeping multi transactions do

Reply:
About the change here. We can avoid allocating strings if we change the moment we capture lock information, I added it as a todo for myself 🤓

Reply:
Right :)

> Using string_view's there won't do the job now

By _now_, do you mean that something has changed?

Reply:
No, I meant for your PR once you capture it after the command has already run